### PR TITLE
i18n: Patterns: Disambiguate singular & plural uses of 'Synced' & 'Unsynced'

### PR DESCRIPTION
**Pull request by mcsf**
## What?
Add or amend i18n context to labels "Synced" and "Unsynced" to disambiguate strings where the adjective qualifies a singular noun ("a [synced] pattern") from those qualifying a plural noun ("the [synced] patterns").

## Why?
English spoils developers by being so lightly inflected. :)

In many languages, adjectives have case inflections for number. In my usual locale (pt-PT):
* o padrão **sincronizado** (_the synced pattern_)
* os padrões **sincronizados** (_the synced patterns_)

Without clear context attached to short strings like `'Synced'`, translators have no way to know which case to use, nor to encode different translations (one for singular and one for plural). For instance, in my locale, I spotted the sync status badge for patterns "mis-numbered":

sync-state-number-mismatch

## How?
Consistently provide i18n context to the strings "Synced" and "Unsynced" whenever they appear in UI to refer to patterns' sync status. This context (via `_x()`) should read:
* `pattern (singular)` when the string is used in the context of a single pattern (e.g. a a status badge in a Data Views grid listing; a sync option when creating a new pattern)
* `patterns` (preexisting context) when the string is used in the context of a group of patterns (e.g. a select control to filter between "All" / "Synced" / "Unsynced")

Previously, strings either had no context or had one that would lead translators to assume the wrong number.

## Context

- #55935: Fix addressing a similar report during the 6.4 cycle, which saw the first addition of i18n context
- #60284: Introduction of badge fields in Data Views

### CODE REVIEW

File: packages/reusable-blocks/src/components/reusable-blocks-menu-items/index.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ import { BlockSettingsMenuControls } from '@wordpress/block-editor';
 import ReusableBlockConvertButton from './reusable-block-convert-button';
 import ReusableBlocksManageButton from './reusable-blocks-manage-button';
 
+/*
+ * FIXME: CAN WE GET RID OF THIS COMPONENT? AS FAR AS I CAN SEE, IT'S EXPORTED
+ * BUT NEVER USED, ONLY MENTIONED IN DOCUMENTATION.
+ */
 export default function ReusableBlocksMenuItems( { rootClientId } ) {
```
Comment (youknowriad) [inline packages/reusable-blocks/src/components/reusable-blocks-menu-items/index.js]: It seems like the whole "reusable-blocks" package can be deprecated. Might be good to do that in its own PR.

Comment (mcsf) [reply packages/reusable-blocks/src/components/reusable-blocks-menu-items/index.js]: > It seems like the whole "reusable-blocks" package can be deprecated. Might be good to do that in its own PR.

Yeah, I added this as a separate commit to easily revert — I just didn't want to forget

**Comment by mcsf** [comment]
Also cc @oandregal for visibility, since you worked a lot on the filters:

One thing I considered for a moment was to generalise this problem at the field level. Specifically, we could require that the possible states of an enumeration include a singular and a plural label. But I've dismissed it for now: in practice, until we have more use cases, it seems like defaulting to singular is enough. WDYT?

**Comment by ellatrix** [comment]
Should this be backported? If so, please add the `Backport to WP Beta/RC` label. Thanks!

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by ellatrix** [comment]
This was cherry-picked to the wp/6.6 branch.
